tracer/src/Datadog.Trace/Tagging/ITagProcessor.cs

```diff
@@ -0,0 +1,14 @@
+// <copyright file="ITagProcessor.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.Tagging
+{
+    internal interface ITagProcessor
+    {
+        void ProcessMeta(ref string key, ref string value);
+
+        void ProcessMetric(ref string key, ref double value);
```

Comment:
Something I think I want to call them all "tags" and use overloads for string tags vs numeric tags, since the meta/metric difference is just a MessagePack detail, but... meh.

```
internal interface ITagProcessor
{
        void ProcessTag(ref string key, ref string value);
        void ProcessTag(ref string key, ref double value);
}
```

Reply:
If we will call them "tags" then I think the name of the interface is enough and we should rename it to `Process` ?

Reply:
I dunno, we have the difference between metrics and tags everywhere else in the code base, I'm inclined to keep it as-is

Reply:
Ok, let's keep it as it in this PR, and if we want to change it, let's create a PR for the whole project.